### ISSUE TITLE
I believe the RSSI function is returning the wrong number.

### ISSUE THREAD

**Issue by nlbutts** [adafruit/Adafruit_CircuitPython_RFM9x#31]
On line 553 the RSSI register (for Lora) is read and 137 is subtracted from the value. During testing I seem to lose the LORA signal at -101 dBm. Looking at the Semtech data sheet (section 5.5.5) the formula for RSSI is:
` RSSI (dBm) = -157 + Rssi, (when using the High Frequency (HF) port)
or
RSSI (dBm) = -164 + Rssi, (when using the Low Frequency (LF) port)
` 

Working the numbers back when this library reads -101, the register would have contained 36, running the correct formula gives me an RSSI of -121 dBm. I'm running the Lora radio at 125 KHz bandwidth and a SF of 7, which should give me a theoretical -125 dBm receive sensitivity. This seems in the realm of possibility. 

Am I missing something or is the number incorrect in the code?

**Comment by brentru** [comment]
@nlbutts
The library code uses RFM9x Datasheet, which seems like it doesn't take HF/LF into account like the SX1276 formula (https://www.digikey.com/en/datasheets/rf-solutions/rf-solutions-rfm95_96_97_98w):

![Microsoft_Word_-_RFM95_96_97_98W_doc](https://user-images.githubusercontent.com/322428/69660883-4851c280-104f-11ea-9ebc-b1a2efb89117.png)

`        return self._read_u8(_RH_RF95_REG_1A_PKT_RSSI_VALUE) - 137`

I'd be open to implementing the RSSI formulas from the SX datasheet in this library, unless you'd want to take this on and open a PR?

**Comment by nlbutts** [comment]
It appears Hoperf fixed the datasheet. I just downloaded Version 2.0 of the datasheet from their web-site and it shows the same formula as Semtech's datasheet. 

Excuse my ignorance, but what is a PR?

**Comment by brentru** [comment]
@nlbutts Cool - confirmed the latest hoperf DS reflects the semtech formulas. (https://www.hoperf.com/data/upload/portal/20190801/RFM95W-V2.0.pdf).

PR as in a pull request to this repository. Alternatively I can update the code and library to reflect the issue

**Comment by kattni** [comment]
@brentru Is this something you're still working on?

**Comment by nlbutts** [comment]
I wanted to use the SX127x parts in both LoRa and FSK mode, so I switched to the UPM code base.

**Comment by brentru** [comment]
@jerryneedell  Was this ever implemented in any of the work you've done for RFM9x?

**Comment by jerryneedell** [comment]
@brentru I have not but it look like a good  thing to do -- there are also some proposed changes in #51 that I have been meaning to test and implement. Unfortunately, I have not had much time available recently. That should be improving this week.
If you want this done immediately, then you are probably better off doing it yourself. I hope to be able to get to this later this week or next week.

**Comment by jerryneedell** [comment]
looks like this will work...unless I am missing somethng.
```
    @property
    def rssi(self):
        """The received strength indicator (in dBm) of the last received message."""
        # Read RSSI register and convert to value using formula in datasheet.
        # Remember in LoRa mode the payload register changes function to RSSI!
        rssi = self._read_u8(_RH_RF95_REG_1A_PKT_RSSI_VALUE)
        if self.low_frequency_mode:
            rssi -= 157
        else:
            rssi -= 164
        return rssi
```

**Comment by brentru** [comment]
@jerryneedell  Yeah, checked the v2 of the hoperf datasheet and the code in this library looks correct to me.

**Comment by jerryneedell** [comment]
It has not been updated yet

**Comment by jerryneedell** [comment]
@brentru I'm confused. Are you saying it does not need to be changed?

**Comment by brentru** [comment]
@jerryneedell  Sorry - I misread your post as a code snippet from `master`! The code you posted looks good & matches the Semtech and hoperf datasheets.

**Comment by jerryneedell** [comment]
@brentru I was planning to combine this with #51. Is that OK or would you prefer separate PR's?

**Comment by brentru** [comment]
@jerryneedell  That's OK - fold it in!

**Comment by jerryneedell** [comment]
fixed by #57
